### PR TITLE
[SYCL][DeviceSanitizer] Add "asan_libdevice.hpp" into cmake of libdevice

### PR DESCRIPTION
**Pull request by AllanZyne**

### CODE REVIEW

File: libdevice/cmake/modules/SYCLLibdevice.cmake [hunk 1]
```diff
@@ -118,7 +118,12 @@ set(imf_obj_deps device_imf.hpp imf_half.hpp imf_bf16.hpp imf_rounding_op.hpp im
 set(itt_obj_deps device_itt.h spirv_vars.h device.h sycl-compiler)
 set(bfloat16_obj_deps sycl-headers sycl-compiler)
 if (NOT MSVC)
-  set(sanitizer_obj_deps device.h atomic.hpp spirv_vars.h include/sanitizer_device_utils.hpp include/spir_global_var.hpp sycl-compiler)
+  set(sanitizer_obj_deps
+    device.h atomic.hpp spirv_vars.h
+    include/asan_libdevice.hpp
+    include/sanitizer_device_utils.hpp
```
[the remark this answers is not in the document]
Comment (AllanZyne) [reply libdevice/cmake/modules/SYCLLibdevice.cmake:124]: I think I used wrong title. Let me correct it.

**Review by bso-intel — APPROVED** [review]

**Comment by AllanZyne** [comment]
Can anyone help to merge this PR?
Thanks very much!

**Comment by AllanZyne** [comment]
I think the ci failure is not related to my PR.
Can gate keeper help to confirm and merge this PR?
Thank you very much!

**Comment by AllanZyne** [comment]
@intel/llvm-gatekeepers please merge.
Thanks.

**Comment by steffenlarsen** [comment]
Looks like failure was only in post-build cleanup on Windows, so it should be fine to proceed.

@AllanZyne - For future reference, if there is a failure in CI it is the author's responsibility to analyze and reason about the failures, assuming they have access to the logs.

**Pull request merged**